### PR TITLE
chore: get rid of legacy agentConfiguration -> actionConfiguration relationship

### LINKED ISSUE #4594: Invert AgentConfiguration and {Action}Configuration relationships (from 1-1 to 1-MANY)
Reference: https://www.notion.so/dust-tt/Multi-Actions-Technical-Proposal-662aa8afbc584f4285d89becc0a6417b?pvs=4

High risk PR that will touch the way we load `{Action}Configurations` for an `AgentConfiguration` (based on a new `agentConfigurationId` on `{Action}Configuration` (vs a `{action}ConfigurationId` on `AgentConfiguration` today, to be removed))

### PR DESCRIPTION
**Pull request by fontanierh**
## Description

fixes https://github.com/dust-tt/dust/issues/4594 (last step)

We now fully rely on the new 1:many relationship. The legacy one was kept around and shadow-written during migration time.
We can now get rid of it.

The 3 action config models had been moved into agent.ts, as we had a relationship both way (so we needed them defined in the same file to avoid circular dependency issues). They have been moved back where they belong (each in their respective file)

## Risk

Should be a functional no-op, and is well tested. But ofc this touches some critical models and includes some destructive schema changes so the blast radius is huge.

## Deploy Plan

Deploy code and then initdb.

### CODE REVIEW

**Review by PopDaph — APPROVED** [review]
LGTM

File: front/lib/models/assistant/actions/dust_app_run.ts [hunk 1]
```diff
@@ -1,13 +1,80 @@
 import type { DustAppParameters } from "@dust-tt/types";
 import type {
   CreationOptional,
+  ForeignKey,
   InferAttributes,
   InferCreationAttributes,
 } from "sequelize";
 import { DataTypes, Model } from "sequelize";
 
+import { AgentConfiguration } from "@app/lib/models/assistant/agent";
```
Comment (spolu) [inline front/lib/models/assistant/actions/dust_app_run.ts:10]: let's use the default model import path

File: front/lib/models/assistant/actions/retrieval.ts [hunk 2]
```diff
@@ -8,10 +8,130 @@ import type {
 } from "sequelize";
 import { DataTypes, Model } from "sequelize";
 
-import { AgentRetrievalConfiguration } from "@app/lib/models/assistant/agent";
+import { AgentConfiguration } from "@app/lib/models/assistant/agent";
```
Comment (spolu) [inline front/lib/models/assistant/actions/retrieval.ts:11]: let's use the default model import path

File: front/lib/models/assistant/actions/tables_query.ts [hunk 3]
```diff
@@ -6,9 +6,64 @@ import type {
 } from "sequelize";
 import { DataTypes, Model } from "sequelize";
 
-import { AgentTablesQueryConfiguration } from "@app/lib/models/assistant/agent";
+import { AgentConfiguration } from "@app/lib/models/assistant/agent";
```
Comment (spolu) [inline front/lib/models/assistant/actions/tables_query.ts:9]: same here

**Review by spolu — APPROVED** [review]
LGTM

**Review by philipperolet — APPROVED** [review]
LGTM

File: front/lib/models/assistant/actions/dust_app_run.ts (hunk 1 above)
Comment (fontanierh) [reply front/lib/models/assistant/actions/dust_app_run.ts]: 🙏

Comment (fontanierh) [reply front/lib/models/assistant/actions/dust_app_run.ts]: Actually... doing this creates a circular dependency issue. I'd be in favor of deleting this index file, as it just tends to create such problems.

**Pull request merged**